File: tests/expectations/both/nonnull.c
#include <stdint.h>
#include <stdlib.h>
#include <stdbool.h>

typedef struct Opaque Opaque;

typedef struct Foo_u64 {
  float *a;
  uint64_t *b;
  Opaque *c;
  uint64_t **d;
  float **e;
  Opaque **f;
  uint64_t *g;
  int32_t *h;
  int32_t **i;
} Foo_u64;

void root(int32_t *arg, Foo_u64 *foo, Opaque **d);

File: tests/expectations/nonnull.zig
const std = @import("std");

const Opaque = opaque {};

pub const Foo_u64 = extern struct {
    _a: ?*f32,
    _b: ?*u64,
    _c: ?*Opaque,
    __d: ?*u64,
    __e: ?*f32,
    __f: ?*Opaque,
    _g: ?*u64,
    _h: ?*i32,
    __i: ?*i32,
};

extern fn root(_arg: ?*i32, _foo: ?*Foo_u64, __d: ?*Opaque) anyopaque;
